common/source/CipherEngine.cpp

```diff
@@ -0,0 +1,227 @@
+//Local Include
+#include "common/CipherEngine.hpp"
+
+// Third party includes
+#include <openssl/conf.h>
+#include <openssl/evp.h>
+#include <openssl/rand.h>
+#include <openssl/err.h>
+
+// Standard includes
+#include <iostream>
+#include <memory>
+#include <string.h>
+#include <string>
+#include <vector>
+#include <fstream>
+#include <iomanip>
+using namespace std;
+namespace SDMS{
+
+    void CipherEngine::handleErrors(void)
+    {
+        ERR_print_errors_fp(stderr);
+        abort();
+    }
+    
+    CipherEngine::CipherString CipherEngine::createCipherString()
+    {
+        CipherString cs;
+        return cs;
```

Comment:
```suggestion
        return CipherString();
```

Reply:
Actually this whole method is redundant, because CipherString is public you can just do this. 
```c++
CipherEngine::CipherString cs;
```

Where ever you need it.